main/SQLHelper.cpp

```diff
@@ -5632,6 +5665,22 @@ bool CSQLHelper::GetLastValue(const int HardwareID, const char* DeviceID, const
 	return result;
 }
 
+std::string  CSQLHelper::GetDeviceValue(const char* FieldName, const std::string& Idx)
+{
+	bool result = false;
+	std::vector<std::vector<std::string> > sqlresult;
+
+    sqlresult = safe_query("SELECT %s FROM DeviceStatus WHERE ( ID= %s )" , FieldName, Idx.c_str() );
+
+	if (!sqlresult.empty())
+	{
+		return sqlresult[0][0] ;
+	}
+
+	return "" ;
```

Comment:
Do it the modern and efficient way
```suggestion
	return {};
```

Reply:
or the readable way 😁

` return std::string("")`

Reply:
Ok done.

Reply:
> or the readable
> 
> ` return std::string("")`

I guess what you mean by readable is the default constructor `return std::string()` or `return std::string{}`. Otherwise `std::string("")` will have to look at the C-string you pass before constructing the `std::string`, which is a waste of time and memory. But probably not the critical hot path in Domoticz for performance, I agree. :-)